### PR TITLE
933 mouseover choropleth regions show popup

### LINKED ISSUE #933: Mouseover Choropleth Regions (i.e., Brushing) Should Show Pop-up
Rather than having to explicitly clicking on a region, I would rather simply be able to mouse over (or brush) a region and see the pop-up that shows region name, completion rate, and asks if I want to audit it.

Would be much easier to identify regions and areas in need with this brushing action...

image

### PR DESCRIPTION
**Pull request by jstadeusiak**
Resolves #933 

When mousing over regions of the choropleth the popup would display as expected. However upon mousing over the popup, the mouseout event was triggered and the black highlighting around the region was removed. To prevent the black highlighting around the region from removing itself I used the event to get the class name triggering the mouseout event, and if that class name did not match any of the potential popup classes, then the highlight clearing function runs. 

Before:
Mouse on popup -
image
Mouse on region - 
image

After: 
Mouse on popup/region - 
image

### CODE REVIEW

File: public/javascripts/AccessibilityChoropleth.js [hunk 1]
```diff
@@ -179,13 +179,23 @@ function AccessibilityChoropleth(_, $, turf, difficultRegionIds) {
...
                 this.setStyle({opacity: 1.0, weight: 3, color: "#000"});
-                this.openPopup()
+                this.openPopup();
             });
             layer.on('mouseout', function (e) {
-                for (var i = layers.length - 1; i >= 0; i--) {
-                    if (currentLayer !== layers[i])
-                        layers[i].setStyle({opacity: 0.25, weight: 1});
+                var mouseoutTrigger = e.originalEvent.toElement.className;
+                if (mouseoutTrigger != "leaflet-popup-content" && mouseoutTrigger != "leaflet-popup-content-wrapper" && mouseoutTrigger != "region-selection-trigger") {
```
Comment (misaugstad) [inline public/javascripts/AccessibilityChoropleth.js:192]: And are all three of those class names options? Is it sometimes one and sometimes the other? Just curious where all those different class names come from.

Comment (jstadeusiak) [reply public/javascripts/AccessibilityChoropleth.js:192]: leaflet-popup-content-wrapper is the whole popup container which can be triggered depending on where the mouse is. leaflet-popup-content is slightly inside that and can also be triggered if its inside the wrapper. region-selection-trigger is the link that is within each region, which can also be triggered if the mouse goes directly to that point.

[the remark this answers is not in the document]
Comment (jstadeusiak) [reply public/javascripts/AccessibilityChoropleth.js:192]: Got it!

[the remark this answers is not in the document]
Comment (jstadeusiak) [reply public/javascripts/AccessibilityChoropleth.js:192]: Switched up to the format you suggested below.

File: public/javascripts/AccessibilityChoropleth.js [hunk 2]
```diff
@@ -179,13 +179,23 @@ function AccessibilityChoropleth(_, $, turf, difficultRegionIds) {
             layers.push(layer);
 
             layer.on('mouseover', function (e) {
+                for (var i = layers.length - 1; i >= 0; i--) {
```
[the remark this answers is not in the document]
Comment (jstadeusiak) [reply public/javascripts/AccessibilityChoropleth.js:182]: Done!

File: public/javascripts/AccessibilityChoropleth.js (hunk 1 above)
Comment (misaugstad) [reply public/javascripts/AccessibilityChoropleth.js:192]: Nice, thank you!

**Comment by misaugstad** [comment]
@jstadeusiak I think you may have forgotten to push your commits that addressed the comments?

**Review by misaugstad — APPROVED** [review]
In the interest of time, and since you've started school, I rewrote some stuff to finish up this PR. The brush functionality was still a little messed up, and we wanted to have the popup go away when the user takes their mouse off of the region. Please feel free to take a look at my edits so you can see how I did it. I'm also down to talk about it sometime if you'd like.

**Pull request merged**